userland/libtock/ble.h

```diff
@@ -2,125 +2,120 @@
+  GAP_COMPLETE_LIST_16BIT_SERVICE_IDS    = 0x03, /* Complete list of 16-bit Service IDs. */
+  GAP_INCOMPLETE_LIST_32BIT_SERVICE_IDS  = 0x04, /* Incomplete list of 32-bit Service IDs (not relevant for Bluetooth 4.0). */
+  GAP_COMPLETE_LIST_32BIT_SERVICE_IDS    = 0x05, /* Complete list of 32-bit Service IDs (not relevant for Bluetooth 4.0). */
+  GAP_INCOMPLETE_LIST_128BIT_SERVICE_IDS = 0x06, /* Incomplete list of 128-bit Service IDs. */
+  GAP_COMPLETE_LIST_128BIT_SERVICE_IDS   = 0x07, /* Complete list of 128-bit Service IDs. */
+  GAP_SHORTENED_LOCAL_NAME               = 0x08, /* Shortened Local Name. */
+  GAP_COMPLETE_LOCAL_NAME                = 0x09, /* Complete Local Name. */
+  GAP_TX_POWER_LEVEL                     = 0x0A, /* TX Power Level (in dBm). */
+  GAP_DEVICE_ID                          = 0x10, /* Device ID. */
+  GAP_SLAVE_CONNECTION_INTERVAL_RANGE    = 0x12, /* Slave Connection Interval Range. */
+  GAP_LIST_128BIT_SOLICITATION_IDS       = 0x15, /* List of 128 bit service UUIDs the device is looking for. */
+  GAP_SERVICE_DATA                       = 0x16, /* Service Data. */
+  GAP_APPEARANCE                         = 0x19, /* Appearance */
+  GAP_ADVERTISING_INTERVAL               = 0x1A, /* Advertising Interval. */
+  GAP_MANUFACTURER_SPECIFIC_DATA         = 0xFF  /* Manufacturer Specific Data. */
+} GapAdvertisementData_t;
+
+enum {
+  LE_LIMITED_DISCOVERABLE = 0x01, /* Peripheral device is discoverable for a limited period of time. */
+  LE_GENERAL_DISCOVERABLE = 0x02, /* Peripheral device is discoverable at any moment. */
+  BREDR_NOT_SUPPORTED     = 0x04, /* Peripheral device is LE only. */
+  SIMULTANEOUS_LE_BREDR_C = 0x08, /* Not relevant - central mode only. */
+  SIMULTANEOUS_LE_BREDR_H = 0x10 /*Not relevant - central mode only. */
+} GapFlags_t;
+
+
+
+
+/*******************************************************************************
+ *   User-space API
+ ******************************************************************************/
+
+
+ // initialize advertisement (should be used before invoken ble_start_advertising)
+ //
+ // advertising_itv_ms  - advertisement interval in milliseconds
+ // flags               - flags in the advertisement defined in GapFlags_t enum
+ // addr                - address to be used in the advertisment
+ // addr_len            - length of the address (it should be 6)
+int ble_initialize(uint16_t advertising_itv_ms, uint8_t flags, uint8_t *addr, uint8_t addr_len);
+
+// start advertising
+int ble_start_advertising(void);
+
+
+// stop advertising but don't change anything in the packet configuration
+int ble_stop_advertising(void);
+
+// reset the entire advertisement data packet
+// excluding flags and advertisement address
+int ble_reset_advertisement(void);
+
+
+ // configure advertisement name
+ //
+ // device_name         - device named to be used in the advertisement
+ // len                 - size in bytes
+int ble_advertise_name(uint8_t *device_name, uint8_t size_b);
```

Comment:
For this class of advertisement functions, what are the possible return values? What value is returned if the advertisement data is now too long to advertise?

Reply:
turns out the "capsule" returned SUCCESS regardless got catch but now it returns ESIZE when the buffer is full.